components/playlist/browser/playlist_service.cc

```diff
@@ -1049,10 +1061,34 @@ bool PlaylistService::GetThumbnailPath(const std::string& id,
 
 bool PlaylistService::GetMediaPath(const std::string& id,
                                    base::FilePath* media_path) {
-  // TODO(sko) The extension of the media file can differ.
-  // We should iterate files in the directory and find the match.
-  *media_path = GetPlaylistItemDirPath(id).Append(
-      PlaylistMediaFileDownloadManager::kMediaFileName);
+  constexpr base::FilePath::CharType kMediaFileName[] =
+      FILE_PATH_LITERAL("media_file");
+
+  std::string extension = "mp4";
+
+  if (HasPlaylistItem(id)) {
+    // Try to infer file extension from the source URL.
+    auto item = GetPlaylistItem(id);
+    GURL url(item->media_source);
+    auto path = url.path();
+    if (!path.empty()) {
+      if (path.at(0) == '/') {
+        // Drop the leading '/'.
+        path = std::string(path.begin() + 1, path.end());
+      }
```

Comment:
I think we don't need this because below code only splits with `.` and only takes last part?

Reply:
Yeah, make sense. Removed.